Cubical/Homotopy/WhiteheadsLemma.agda

```diff
@@ -0,0 +1,447 @@
+   ≡ cong f
+ΩCongSquare f =
+  J ( λ b p → (λ q → q ∙ (cong f p))
+             ∘ fst (Ω→ (f , refl))
+             ∘ (λ q → q ∙ sym p)
+             ≡ cong f)
+    ( funExt (λ x → sym (rUnit _)
+                   ∙ sym (doubleCompPath-filler-refl _)
+                   ∙ cong (cong f) (sym (rUnit _))))
+
+private
+  setMapFunctorial : {A B C : Type ℓ} (f : A → B) (g : B → C)
+    → map g ∘ map f ≡ map (g ∘ f)
+  setMapFunctorial f g =
+    funExt (sElim (λ x → isSetPathImplicit) λ a → refl)
+
+  πHomπHomCongSquareAux : {A B : Type ℓ} {a : A} {n : ℕ} (f : A → B)
+    → Iso.inv (Iso-πΩ-π (1 + n))
+     ∘ (fst (πHom {A = (A , a)} {B = (B , f a)} (1 + n) (f , refl)))
+     ∘ (Iso.fun (Iso-πΩ-π (1 + n)))
+     ≡ map (Iso.inv (invIso (flipΩIso (1 + n)))
+         ∘ fst ((Ω^→ (2 + n)) (f , refl))
+         ∘ Iso.fun (invIso (flipΩIso (1 + n))))
+  πHomπHomCongSquareAux {n = n} f =
+      cong
+      ( λ g → g ∘ (Iso.fun (Iso-πΩ-π (1 + n))))
+      ( setMapFunctorial
+        ( fst ((Ω^→ (2 + n)) (f , refl)))
+        ( Iso.fun (flipΩIso (1 + n))))
+    ∙ setMapFunctorial
+      ( Iso.inv (flipΩIso (1 + n)))
+      ( Iso.fun (flipΩIso (1 + n)) ∘ fst ((Ω^→ (2 + n)) (f , refl)))
+
+-- Ω (Ω→ f) ≡ Ω→ (Ω f)
+PathPΩ^→Ω : {A B : Pointed ℓ} (n : ℕ) (f : A →∙ B)
+  → PathP (λ i → (flipΩPath {A = A} n) i
+                   →∙ (flipΩPath {A = B} n) i)
+           ((Ω^→ (1 + n)) f) ((Ω^→ n) (Ω→ f))
+PathPΩ^→Ω zero f = refl
+PathPΩ^→Ω (suc n) f =
+  specialCong∙ Ω Ω→
+  ( flipΩPath n)
+  ( flipΩPath n)
+  ( Ω^→ (1 + n) f)
+  ( (Ω^→ n) (Ω→ f))
+  ( PathPΩ^→Ω n f)
+
+private
+  flipIsoSquare : {A B C D : Type ℓ} (f : A → B) (g : C → D)
+    (H : Iso A C) (J : Iso B D)
+    → Iso.inv J ∘ g ∘ (Iso.fun H) ≡ f
+    → Iso.inv (invIso J) ∘ f ∘ Iso.fun (invIso H) ≡ g
+  flipIsoSquare f g H J p =
+    sym
+    ( funExt
+      ( λ x →
+          sym ((Iso.rightInv J) (g x))
+          ∙ cong (λ y → Iso.fun J (Iso.inv J (g y)))
+                 (sym ((Iso.rightInv H) x))
+          ∙ cong (λ h → (Iso.fun J ∘ h ∘ Iso.inv H) x) p))
+
+{-
+
+Ωⁿ⁺¹ (Ω (A, a)) -------> Ωⁿ⁺¹ (Ω (B, f a))
+        |                        ^
+        |                        |
+        |                        |
+        v                        |
+Ω (Ωⁿ⁺¹ (A, a)) -------> Ω (Ωⁿ⁺¹ (B, f a))
+
+-}
+Ω+1ΩCongSquare : {A B : Type ℓ} {a : A} {n : ℕ} (f : A → B)
+  → Iso.inv (invIso (flipΩIso (1 + n)))
+   ∘ fst ((Ω^→ {A = (A , a)} (2 + n)) (f , refl))
+   ∘ Iso.fun (invIso (flipΩIso (1 + n)))
+  ≡ fst ((Ω^→ (1 + n)) (Ω→ (f , refl)))
+Ω+1ΩCongSquare {A = A} {a = a} {n = n} f =
+  flipIsoSquare
+  ( fst ((Ω^→ {A = (A , a)} (2 + n)) (f , refl)))
+  ( fst ((Ω^→ (1 + n)) (Ω→ (f , refl))))
+  ( flipΩIso (1 + n))
+  ( flipΩIso (1 + n))
+  ( ∙PathP→Square
+    ( flipΩPath (1 + n))
+    ( flipΩPath (1 + n))
+    ( (Ω^→ {A = (A , a)} (2 + n)) (f , refl))
+    ( (Ω^→ (1 + n)) (Ω→ (f , refl)))
+    ( PathPΩ^→Ω (1 + n) (f , refl)))
+
+private
+  ∙∙Lemma : {A B : Type ℓ} {f : A → B} {a b : A} (p : a ≡ b)
```

Comment:
This one can be removed. If you just use `sym (rUnit (cong f p))` for this one, it is definitionally equal to `∙∙lCancel refl` when p is `refl`